classes/03class/docker/README.md

```diff
@@ -0,0 +1,305 @@
+- [ECS](#ecs)
+  - [ECS Cluster](#ecs-cluster)
+  - [ECS Service](#ecs-service)
+  - [ECS Task Definition](#ecs-task-definition)
+- [ECR](#ecr)
+- [Main Take Aways](#main-take-aways)
+
+## Containers
+
+### What is a container?
+A container is a standard unit of software that packages up code and all its dependencies so the application runs quickly and reliably from one computing environment to another. 
+
+A container is supposed to run a single main process. Once the process is completed/stoppped, the container will exit.
+<p align="center">
+    <img src="assets/container01.jpg" width="900">
+</p>
+
+### Difference between Virtual Machine and Container
+Both the virtual machine and the container works with resource isolation, which protect one of the instances(VM or container) to access resources from another instance, even having both running on the same hardware. 
+
+Despite both having the resouce isolation feature, they work in a different way. While the virtual machine virtualizes the hardware, the container virtualizes the operating system. This makes the container a lot more portable and efficient.
+
+<p align="center">
+    <img src="assets/container02.jpg" width="900">
+</p>
+
+#### Container
+Containers are an abstraction at the app layer that packages code and dependencies together. 
+
+Multiple containers can run on the same machine and share the OS kernel with other containers, each running as isolated processes in user space. Containers take up less space than VMs (container images are typically tens of MBs in size), can handle more applications and require fewer VMs and Operating systems. 
+
+#### Virtual Machine
+Virtual machines (VMs) are an abstraction of physical hardware turning one server into many servers.
+
+A hypervisor is a process that separates a computer’s operating system and applications from the underlying physical hardware. The hypervisor allows multiple VMs to run on a single machine.
+
+Each VM includes a full copy of an operating system, the application, binaries, and libraries - the full size can take in the scale of GBs. 
+
+### Benefits of Containers
+Before talking about the benefits of containers, it's important to make clear that not all kinds of workloads are a good fit for a container. Because of the container nature, an application with multiple processes and services running on a single machine is not a good fit for a container. 
+
+Once you confirm that a specific service is a good fit for a container, the benefits of a container includes:
+- Container images are normally small compared to VMs, so it's easy to move them around
+- Because everything required by the application is inside the container, the execution of the container will be the same anywhere
+- Speeds up the development process since the Developer can have multple containers running on his local computer to simulate a very reliable production environment(in terms of funcionality)
```

Comment:
```suggestion
- Speeds up the development process since the Developer can have multple containers running on his local computer to simulate a very reliable production environment (in terms of funcionality)
```